app/src/main/java/org/dieschnittstelle/mobile/android/skeleton/MainActivity.java

```diff
@@ -1,34 +1,98 @@
+import org.dieschnittstelle.mobile.android.skeleton.viewmodel.TaskListViewModel;
 
 public class MainActivity extends AppCompatActivity {
 
     private TextView welcomeText;
-    private FloatingActionButton showTaskListAction;
+    private EditText emailAdress;
+    private EditText password;
+    private Button showTaskListAction;
+    private MainViewModel viewModel;
+    private IUserDatabaseOperation userDBOperation;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        viewModel = new ViewModelProvider(this).get(MainViewModel.class);
+
+        ActivityMainBinding MainViewBinding = DataBindingUtil.setContentView(
+                this,
+                R.layout.activity_main
+        );
+        MainViewBinding.setMainViewModel(this.viewModel);
+        MainViewBinding.setLifecycleOwner(this);
+
+        userDBOperation = ((TaskApplication) getApplication()).getUserDatabaseOperation();
+        viewModel.setUserDBOperation(userDBOperation);
+
+        userDbInitialise(); // insert testdata
 
         welcomeText = findViewById(R.id.welcomeText);
         welcomeText.setText(R.string.welcome_message);
-        welcomeText.setOnClickListener(view -> this.showTaskListView());
 
-        showTaskListAction = findViewById(R.id.showTaskListAction);
-        showTaskListAction.setOnClickListener(view -> this.showTaskListView());
+        emailAdress = findViewById(R.id.editTextTextEmailAddress);
+        password = findViewById(R.id.editTextTextPassword);
+
+        // Authentication
+        this.viewModel.isUsernameNotAnEMail().observe(this, onUsernameNotAnEMail -> {
+            if (onUsernameNotAnEMail) {
+                showMessage(getString(R.string.login_username_error),3);
+            }
+        });
+
+        this.viewModel.isUserAuthenticationFailed().observe(this, onUserAuthenticationFailed -> {
+            if (onUserAuthenticationFailed) {
+                showMessage(getString(R.string.login_authentication_failed),3);
+            }
+        });
+
+        this.viewModel.isUserAuthenticated().observe(this, onAuthenticated -> {
+            if (onAuthenticated) {
+                Intent callTaskOverviewIntent = new Intent(this, TaskListViewActivity.class);
+                startActivity(callTaskOverviewIntent);
+                this.finish();
+            }
+        });
+    }
+
+    private void showMessage(String message, int snackbarDuration) {
+        switch (snackbarDuration) {
+            case 1:
+                Snackbar.make(findViewById(R.id.rootView), message, Snackbar.LENGTH_SHORT).show();
+            case 2:
+                Snackbar.make(findViewById(R.id.rootView), message, Snackbar.LENGTH_LONG).show();
+            case 3:
+                Snackbar.make(findViewById(R.id.rootView), message, Snackbar.LENGTH_INDEFINITE).show();
+        }
```

Comment:
die Methode können wir optimieren.

```suggestion

    private void showMessage(String message) {
        Snackbar.make(findViewById(R.id.rootView), message, Snackbar.LENGTH_INDEFINITE).show();
    }
```